bitcoinlib/services/groestlcoind.py

```diff
@@ -0,0 +1,271 @@
+
+        :param configfile: Path to config file. Leave empty to look in default places
+        :type: str
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
+        :type: str
+
+        :return GroestlcoindClient:
+        """
+        try:
+            config = configparser.ConfigParser(strict=False)
+        except TypeError:
+            config = configparser.ConfigParser()
+        config_fn = 'bitcoin.conf'
+        if isinstance(network, Network):
+            network = network.name
+        if network == 'testnet':
+            config_fn = 'bitcoin-testnet.conf'
+
+        cfn = None
+        if not configfile:
+            config_locations = ['~/.bitcoinlib', '~/.bitcoin', '~/Application Data/Bitcoin',
+                                '~/Library/Application Support/Bitcoin']
+            for location in config_locations:
+                cfn = Path(location, config_fn).expanduser()
+                if cfn.exists():
+                    break
+        else:
+            cfn = Path(BCL_DATA_DIR, 'config', configfile)
+        if not cfn or not cfn.is_file():
+            raise ConfigError("Config file %s not found. Please install bitcoin client and specify a path to config "
+                              "file if path is not default. Or place a config file in .bitcoinlib/bitcoin.conf to "
+                              "reference to an external server." % cfn)
+
+        try:
+            config.read(cfn)
+        except Exception:
+            with cfn.open() as f:
+                config_string = '[rpc]\n' + f.read()
+            config.read_string(config_string)
+
+        testnet = _read_from_config(config, 'rpc', 'testnet')
+        if testnet:
+            network = 'testnet'
+        if _read_from_config(config, 'rpc', 'rpcpassword') == 'specify_rpc_password':
+            raise ConfigError("Please update config settings in %s" % cfn)
+        if network == 'testnet':
+            port = 18332
+        else:
+            port = 8332
+        port = _read_from_config(config, 'rpc', 'rpcport', port)
+        server = '127.0.0.1'
+        server = _read_from_config(config, 'rpc', 'rpcconnect', server)
+        server = _read_from_config(config, 'rpc', 'bind', server)
+        server = _read_from_config(config, 'rpc', 'externalip', server)
+
+        url = "http://%s:%s@%s:%s" % (config.get('rpc', 'rpcuser'), config.get('rpc', 'rpcpassword'), server, port)
+        return GroestlcoindClient(network, url)
+
+    def __init__(self, network='bitcoin', base_url='', denominator=100000000, *args):
+        """
+        Open connection to bitcoin node
+
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
+        :type: str
+        :param base_url: Connection URL in format http(s)://user:password@host:port.
+        :type: str
+        :param denominator: Denominator for this currency. Should be always 100000000 (satoshis) for bitcoin
```

Comment:
:param denominator: Denominator for this currency. Should be always 100000000 (gros) for groestlcoin